components/masonry/Masonry.tsx

```diff
@@ -123,6 +123,56 @@ const Masonry = React.forwardRef<MasonryRef, MasonryProps>((props, ref) => {
     setMergedItems(items || []);
   }, [items]);
 
+  const [viewState, setViewState] = useState({ scrollTop: 0, clientHeight: 0 });
+
+  // ====================== Scroll ======================
+  useEffect(() => {
+    const container = containerRef.current;
+
+    if (!container) return;
+
+    // find scrollable parent
+    let scrollNode: HTMLElement | Window = window;
+    let parent = container.parentElement;
+
+    while (parent) {
```

Comment:
Maybe we can just let scroll ability with the root div instead of parent loop. It will meet too many edge case for this. And `getComputedStyle` is also a perf cost.